modules/ml/src/main/java/org/apache/ignite/ml/math/functions/IgniteDifferentiableDoubleToDoubleFunction.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.math.functions;
+
+/**
+ * Interface for differentiable functions from double to double.
+ */
+public interface IgniteDifferentiableDoubleToDoubleFunction extends IgniteDoubleFunction<Double> {
+    /**
+     * Get function differential at a given point.
+     *
+     * @param pnt Point to calculate differential at.
+     * @return Function differential at a given point.
+     */
+    double differential(double pnt);
```

Comment:
maybe better to rename into "calcDifferential" or "calcDf": add action to name and shortenize form